src/assisted_test_infra/test_infra/controllers/ipxe_controller/ipxe_controller.py

```diff
@@ -37,9 +40,40 @@ def _build_server_image(self):
     def _download_ipxe_script(self, infra_env_id: str, cluster_name: str):
         log.info(f"Downloading iPXE script to {self._ipxe_scripts_folder}")
         utils.recreate_folder(self._ipxe_scripts_folder, force_recreate=False)
-        self._api_client.download_and_save_infra_env_file(
-            infra_env_id=infra_env_id, file_name="ipxe-script", file_path=f"{self._ipxe_scripts_folder}/{cluster_name}"
-        )
+        pxe_content = self._api_client.client.v2_download_infra_env_files(
+            infra_env_id=infra_env_id, file_name="ipxe-script", _preload_content=False
+        ).data.decode('utf-8')
+
+        # PXE can not boot from http redirected to https, update the assets images to local http server
+        if self._local_pxe_assets:
+            pxe_content = self._download_ipxe_assets(pxe_content)
+
+        with open(f"{self._ipxe_scripts_folder}/{cluster_name}", "w") as _file:
+            _file.writelines(pxe_content)
+
+    @staticmethod
+    def _replace_assets_pxe(pxe_content: str, old_asset: str, new_asset: str):
+        log.info(f"Replace pxe assets {old_asset} to {new_asset}")
+        return pxe_content.replace(old_asset, new_asset)
+
+    def _download_ipxe_assets(self, pxe_content: str) -> str:
+        """ Download the ipxe assets to the container http server
+            Update the ipxe-script assets download from the container.
+            The new asset will be downloaded from http://{self._ip}:{self._port}
+            return new updated pxe content.
+        """
+
+        # New pxe content replace http links to local http server
+        new_pxe_content = copy.deepcopy(pxe_content)
+        new_asset = f"http://{self._ip}:{self._port}/"
+        http_to_download = [res for res in pxe_content.split() if "http:" in res]
+        for http in http_to_download:
+            http = http[http.index("http"):]  # in case http not at the beginning
+            for img in ["pxe-initrd", "kernel", "rootfs"]:
+                if img in http:
+                    utils.run_command(f"curl -L '{http}' -o {self._ipxe_scripts_folder}/{img}")
```

Comment:
please use `urllib.request`instead and also add try / except to vlidate the file has been downloaded

Reply:
Ack , thanks.
Already implemented in utiil.download_file